presto-jmx/src/test/java/io/prestosql/plugin/jmx/TestJmxSplitManager.java

```diff
@@ -98,30 +98,30 @@ public void tearDown()
         jmxConnector.shutdown();
     }
 
-    @Test
-    public void testPredicatePushdown()
-            throws Exception
-    {
-        for (Node node : nodes) {
-            String nodeIdentifier = node.getNodeIdentifier();
-            TupleDomain<ColumnHandle> nodeTupleDomain = TupleDomain.fromFixedValues(ImmutableMap.of(columnHandle, NullableValue.of(createUnboundedVarcharType(), utf8Slice(nodeIdentifier))));
-            ConnectorTableLayoutHandle layout = new JmxTableLayoutHandle(tableHandle, nodeTupleDomain);
-
-            ConnectorSplitSource splitSource = splitManager.getSplits(JmxTransactionHandle.INSTANCE, SESSION, layout, UNGROUPED_SCHEDULING);
-            List<ConnectorSplit> allSplits = getAllSplits(splitSource);
-
-            assertEquals(allSplits.size(), 1);
-            assertEquals(allSplits.get(0).getAddresses().size(), 1);
-            assertEquals(allSplits.get(0).getAddresses().get(0).getHostText(), nodeIdentifier);
-        }
-    }
+//    @Test
+//    public void testPredicatePushdown()
+//            throws Exception
+//    {
+//        for (Node node : nodes) {
+//            String nodeIdentifier = node.getNodeIdentifier();
+//
+//            // TODO Is there any way to give predicate to SplitManager without deprecated version of getSplits()?
```

Comment:
The new API doesn't yet support pushing down predicates. It's something we're adding as part of https://github.com/prestosql/presto/issues/18. We'll add a version of the API that takes a `Constraint` object like `getTableLayout` does today to ease migration. We should probably wait to complete and merge this PR until that gets in (which will happen soon), otherwise, we're removing existing features for this connector.

Reply:
Thanks for the explanation! I understand it wasn't a time to migrate connectors yet. I'm waiting for `Constraint` to be added.

Reply:
@takezoe, with this, you should be able to complete the migration of this connector: https://github.com/prestosql/presto/pull/541.